lib/web3-contracts.js

```diff
@@ -131,248 +106,189 @@ export function useTokenBalance(symbol) {
-      setBalance(bigNum(-1))
-      return
-    }
-
-    cancelBalanceUpdate.current = () => {
-      cancelled = true
-    }
-
-    jurorsRegistryContract.balanceOf(account).then(([activeBalance]) => {
-      if (!cancelled) {
-        setBalance(activeBalance)
-      }
-    })
-  }, [account, tokenContract, jurorsRegistryContract])
+  const { loading: supplyLoading, totalSupply } = useAnjTotalSupply()
 
   useEffect(() => {
-    // Always update the balance if updateBalance() has changed
-    updateBalance()
-
-    if (!wrapperContract || !account) {
+    let cancelled = false
+    let retryTimer
+
+    if (
+      antTreasuryBalance.eq(-1) ||
+      !bancorContract ||
+      supplyLoading ||
+      totalSupply.eq(-1)
+    ) {
       return
     }
-
-    const onBought = (from, to, value) => {
-      if (from === account) {
-        updateBalance()
+    const getSalePrice = async () => {
+      try {
+        setLoading(true)
+        const salePrice = await (forwards
+          ? bancorContract.calculatePurchaseReturn(
+              totalSupply,
+              antTreasuryBalance,
+              CONNECTOR_WEIGHT,
+              amount
+            )
+          : bancorContract.calculateSaleReturn(
+              totalSupply,
+              antTreasuryBalance,
+              CONNECTOR_WEIGHT,
+              amount
+            ))
+        if (!cancelled) {
+          setLoading(false)
+          setPrice(salePrice)
+        }
+      } catch (err) {
+        if (!cancelled) {
+          retryTimer = setTimeout(useBondingCurvePrice, RETRY_EVERY)
+        }
       }
     }
-    wrapperContract.on('Bought', onBought)
+
+    getSalePrice()
 
     return () => {
-      wrapperContract.removeListener('Bought', onBought)
+      cancelled = true
+      clearTimeout(retryTimer)
     }
-  }, [account, wrapperContract, updateBalance])
-
-  return balance
+  }, [
+    amount,
+    antTreasuryBalance,
+    bancorContract,
+    forwards,
+    supplyLoading,
+    totalSupply,
+  ])
+
+  return useMemo(() => ({ loading, price }), [loading, price])
 }
 
 // Convert ANT to ANJ action
-export function useConvertAntToAnj() {
+export function useOpenOrder() {
+  const { account } = useWeb3Connect()
   const antContract = useKnownContract('TOKEN_ANT')
-  const [wrapperAddress] = getKnownContract('WRAPPER')
+  const fundraisingContract = useKnownContract('FUNDRAISING')
+
+  const [antAddress] = getKnownContract('TOKEN_ANT')
+  const [marketMakerAddress] = getKnownContract('MARKET_MAKER')
 
   return useCallback(
-    async antAmount => {
-      if (!antContract || !wrapperAddress) {
+    async (amount, forwards = true) => {
+      if (!antContract || !fundraisingContract) {
         return false
       }
+      // Interacting with the bonding curve involves 3 transactions (or two,
+      // depending on the direction):
+      // 1. Approval (if there's not enough allowance and we're converting ANT -> ANJ)
```

Comment:
Right now we're sending the approve and open order transactions at the same time, but with the stepper, we'll  want to wait between the two.

Reply:
With the UI work we'll break up this function into two, so this comment will be split up.